spec/integration/result_spec.rb

```diff
@@ -50,6 +50,33 @@
     end
   end
 
+  describe "#error?" do
+    subject { result }
+
+    let(:schema_result) do
+      double(:schema_result, message_set: [], to_h: {email: "jane@doe.org"})
+    end
+
+    let(:result) do
+      Dry::Validation::Result.new(schema_result) do |r|
+        r.add_error(Dry::Validation::Message.new("root error", path: [nil]))
+        r.add_error(Dry::Validation::Message.new("email error", path: [:email]))
+      end
+    end
+
+    it "reports an error on email" do
+      expect(subject.error?(:email)).to eq true
+    end
+
+    it "reports an error on 'root' when asked for [nil]" do
+      expect(subject.error?([nil])).to eq true
```

Comment:
I added this because it's current behavior (given the root error path that I just copied from the other test). Not sure if it should be like that or if there's a nicer way?

Reply:
Yes base errors are represented by `nil` key, doing `error?(nil)` would work too.

Reply:
I double checked and error?(nil) doesn't work.

```
  1) Dry::Validation::Result#error? reports an error on 'root' when asked for nil
     Failure/Error: errors.any? { |msg| Schema::Path[msg.path].include?(Schema::Path[key]) }
     
     ArgumentError:
       +spec+ must be either a Symbol, Array, Hash or a Path
     # ./lib/dry/validation/result.rb:106:in `block in error?'
     # ./lib/dry/validation/result.rb:106:in `any?'
     # ./lib/dry/validation/result.rb:106:in `error?'
     # ./spec/integration/result_spec.rb:70:in `block (3 levels) in <top (required)>'

Finished in 0.61031 seconds (files took 0.60271 seconds to load)
184 examples, 1 failure

Failed examples:

rspec ./spec/integration/result_spec.rb:69 # Dry::Validation::Result#error? reports an error on 'root' when asked for nil
```

Should it work? Then I'll see how it could be made to work but it seems like that might be in `Schema::Path[key]` then aka in dry-schema :)

Reply:
@PragTob ah right, I forgot that `Path` doesn't turn `nil` into `[nil]` automatically, so yes, this is expected. BTW - for base errors (aka root errors), we have `Result#base_error?` method.